### ISSUE TITLE
`scheduled_at` option isn't being respected

### ISSUE THREAD

**Issue by lucasmazza** [sorentwo/oban#70]
Following the `scheduled_at` example on the README insert jobs as `completed` rather than `scheduled`, while `schedule_in` results in the expected behaviour. I first suspected this after trying out on our app by reading the integration tests on https://github.com/sorentwo/oban/blob/master/test/integration/scheduling_test.exs and noticing that the test is only exercising the `schedule_in` option

```elixir
%{id: 1}
|> MyApp.Business.new(scheduled_at: ~U[2020-12-25 19:00:56.0Z])
|> Oban.insert()
# job will be `completed` with a `completed_at` at now and with the correct `scheduled_at` column

%{id: 1}
|> MyApp.Business.new(schedule_in: 5)
|> Oban.insert()
# job will be `scheduled` and with a null `completed_at` column
```

**Comment by sorentwo** [comment]
This sounds _slightly_ incorrect, though there is definitely a bug here. I expect that the job is being inserted as `available`, but it is running immediately because the state isn't `scheduled`.

Thanks for the report though! Fixed with 827969e

**Comment by lucasmazza** [comment]
@sorentwo thanks for the quick reply and fix! 💜
